EIPS/eip-1613.md

```diff
@@ -67,15 +67,19 @@ Implementing a `RelayRecipient` contract:
 * Maintain a small balance of ETH gas prepayment deposit in `RelayHub`. Can be paid directly by the `RelayRecipient` contract, or by the dapp's owner on behalf of the `RelayRecipient` address. 
   The dapp owner is responsible for ensuring sufficient balance for the next transactions, and can stop depositing if something goes wrong, thus limiting the potential for abuse of system bugs. In DAO usecases it will be up to the DAO logic to maintain a sufficient deposit.
 * Use `get_sender()` and `get_data()` instead of `msg.sender` and `msg.data`, everywhere. `RelayRecipient` provides these functions and gets the information from `RelayHub`.
-* Implement a `may_relay(address relay, address from, uint32 gasPrice, bytes transaction)` view function that returns `true` if and only if it is willing to accept a transaction and pay for it. 
-  `may_relay` is called by `RelayHub` as a view function when a `Relay` inquires it, and also during the actual transaction. Transactions are reverted if `false`, and `Relay` only gets compensated for transactions (whether successful or reverted) if `may_relay` returns `true`. Some examples of `may_relay()` implementations:
+* Implement a `accept_relayed_call(address relay, address from, uint32 gasPrice, bytes encoded_function)` view function that returns **zero** if and only if it is willing to accept a transaction and pay for it. 
```

Comment:
Function signature doesn't match the current implementation (which currently doesn't pass gasPrice). See our discussion in the implementation PR, re gas_price and transaction_fee to facilitate the decision.

Reply:
removed, for now..